code/modules/research/xenobiology/crossbreeding/regenerative.dm

```diff
@@ -9,39 +14,118 @@ Regenerative extracts:
 	effect = "regenerative"
 	icon_state = "regenerative"
 
-/obj/item/slimecross/regenerative/proc/core_effect(mob/living/carbon/human/target, mob/user)
-	return
-/obj/item/slimecross/regenerative/proc/core_effect_before(mob/living/carbon/human/target, mob/user)
-	return
+	var/static/list/bodypart_replacement_table = list(
+		/obj/item/bodypart/arm/right = /obj/item/bodypart/arm/right/slime,
+		/obj/item/bodypart/arm/left = /obj/item/bodypart/arm/left/slime,
+		/obj/item/bodypart/leg/right = /obj/item/bodypart/leg/right/slime,
+		/obj/item/bodypart/leg/left = /obj/item/bodypart/leg/left/slime,
+		/obj/item/bodypart/head = /obj/item/bodypart/head/slime,
+		/obj/item/bodypart/chest = /obj/item/bodypart/chest/slime,
+	)
+
+/obj/item/slimecross/regenerative/proc/core_effect(mob/living/target, mob/user)
+	var/visible_message = "[user] crushes [src] over [target], the milky goo quickly regenerating all of [target.p_their()] injuries!"
+	var/self_message = "You squeeze [src], and it bursts over [target], the milky goo regenerating [target.p_their()] injuries."
+
+	var/obj/item/bodypart/type_to_instantiate
+	var/obj/item/bodypart/targetted_part
+	if (ishuman(target))
+		var/mob/living/carbon/human/human_target = target
+		targetted_part = get_bodypart_to_replace(user, human_target)
+
+		if (!bodypart_valid_for_healing(targetted_part, user, target))
+			return FALSE
+
+		for (var/obj/item/bodypart/iterated_typepath as anything in bodypart_replacement_table)
+			if (istype(targetted_part, iterated_typepath))
+				type_to_instantiate = bodypart_replacement_table[iterated_typepath]
+				break
+
+		if(human_target != user)
+			visible_message = "[user] crushes [src] over [human_target], the milky goo enveloping [human_target.p_their()] [targetted_part.name] and perfecting it into slime!"
+			self_message = "You crush [src] over [human_target], the milky goo enveloping [human_target.p_their()] [targetted_part.name] and perfecting it into slime!"
+		else
+			visible_message = "[user] crushes [src] over [user.p_them()]self, the milky goo enveloping [user.p_their()] [targetted_part.name] and perfecting it into slime!"
+			self_message = "You crush [src] over yourself, the milky goo enveloping your [targetted_part.name] and perfecting it into slime!"
+
+		user.visible_message(span_notice(visible_message), span_notice(self_message))
+
+	do_healing(target, user, type_to_instantiate, targetted_part)
+
+	return TRUE
 
-/obj/item/slimecross/regenerative/afterattack(atom/target,mob/user,prox)
+/obj/item/slimecross/regenerative/proc/get_bodypart_to_replace(mob/living/user, mob/living/carbon/human/target)
+	var/obj/item/bodypart/targetted_part = target.get_bodypart(user.zone_selected)
+	return targetted_part
```

Comment:
Not quite sure why this is its own proc instead of just doing `get_bodypart()`, I would've expected it be overridden or something but no

Reply:
Yeah it's a bit of a fucko pr made by a sleep-deprived brain, I'll spruce it up tmr